7.0 bill of lading webkit

mcassuto opened the pull request:
A Bill of Lading is a document issued by a carrier to a shipper of goods.
It is a negotiable instrument, and it serves three purposes:
- it is a receipt for the goods shipped;
- it evidences the contract of carriage;
- it serves as a document of title (i.e. ownership).

Commit:
bill_of_lading_webkit: initialize module

Commit:
bill_of_lading_webkit: fix PEP8 constraints

Commit:
bill_of_lading_webkit: remove SFL icon

Commit:
bill_of_lading_webkit: fix PEP8 constraints

bwrsandman commented:
`./bill_of_lading_webkit/report/stock_report.py:25:1: F401 'pooler' imported but unused`

Commit:
bill_of_lading_webkit : remove unused import of pooler

Commit:
stock_picking_webkit: fix all PEP8 violations

mcassuto commented:
@bwrsandman : I removed the import of pooler + I fixed also all the PEP8 violation from the other module to get travis successfull

pedrobaeza commented:
Hi, following module naming conventions, I would call this module as _stock_bill_of_lading_webkit.

Regards.

bwrsandman commented:
@mcassuto @pedrobaeza I fixed .travis.yml and added it to this PR

Commit:
Update travis

mcassuto commented:
Hi @pedrobaeza ,

where can I find this naming convention ?
I would like to improve my commits and not having you to remind me this ;-)

bwrsandman commented:
@mcassuto only thing causing failiure is:
`./stock_picking_webkit/report/stock_report.py:116:9: E265 block comment should start with '# '`

pedrobaeza commented:
@mcassuto, thanks for your interest. That's something not written yet, because OCA conventions needs a review, but you can find this convention across all the core modules. The convention is more or less:

```
area_subarea_function
```

where

```
__area__ is the main area affected (stock, sale, mrp, purchase, account...)
__subarea__ is an object or subdivision reference. For example, inventory (for stock), invoice (for account), etc.
__function__ This is a free denomination that must be short and descriptive for the modification that the module is doing.
```

In your case, I think you don't need subarea, because the function (bill of lading) is generic enough.

Regards.

Commit:
Fix the last Pep8 error (this one was not catched by PuCharm)

Commit:
stock_bill_of_lading_webkit: renaming module to respect naming convention

mcassuto commented:
@bwrsandman : I fixed the Pep8 error, but it is a strange situation of having to fix errors on already published modules to get mine merged....
@pedrobaeza : thanks for the explanation , I renamed my module

bwrsandman commented:
:+1:

pedrobaeza commented:
Hi, Marc, one last thing: please rename module name on the translations.

Thanks.

bwrsandman commented:
Adding to that, the .pot file needs to be regenerated once it's renamed:
- [x] stock_bill_of_lading_webkit/i18n/bill_of_lading_webkit.pot -> stock_bill_of_lading_webkit/i18n/stock_bill_of_lading_webkit.pot
- [x] regenerate .pot file
- [x] Update fr.po file

Commit:
stock_bill_of_lading_webkit: renaming the module broke the report

mcassuto commented:
hi @pedrobaeza ,

finally do I have your approval on this one ?

Regards,
Marc

pedrobaeza commented:
Hi, Marc, please fix translations issue, and I'll get you my approval. Put a comment here to get notified when you do the change.

Regards.

Commit:
stock_bill_of_lading_webkit: translations files regenerated after module renaming

mcassuto commented:
@pedrobaeza : my very bad, I understood  Sandy did it.
Fixed !

pedrobaeza commented:
Where is the .pot file?

Commit:
stock_bill_of_lading_webkit : forgot the renamed pot file

mcassuto commented:
@pedrobaeza here ;-)

pedrobaeza commented:
Thanks.

:+1:

bwrsandman commented:
Needs to update fr.po file

pedrobaeza commented:
Are you sure? I see it correct.

bwrsandman commented on stock_bill_of_lading_webkit/i18n/fr.po:
```diff
@@ -0,0 +1,103 @@
+# Translation of OpenERP Server.
+# This file contains the translation of the following modules:
+#	* bill_of_lading_webkit
```
not correct

bwrsandman commented:
:+1:
